consumer-test/consumerTestController.groovy

```diff
@@ -0,0 +1,144 @@
+
+EXCLUDED_FROM_CONSUMER_TESTING_REGEXES = [
+    /^documentation\/.*$/,
+    /^.travis.yml$/,
+    /^test\/.*$/
+]
+
+newEmptyDir(WORKSPACES_ROOT)
+ConsumerTestUtils.workspacesRootDir = WORKSPACES_ROOT
+ConsumerTestUtils.libraryVersionUnderTest = "git log --format=%H -n 1".execute().text.trim()
+ConsumerTestUtils.repositoryUnderTest = System.getenv('TRAVIS_REPO_SLUG') ?: 'SAP/jenkins-library'
+
+def testCaseThreads
+def filePath = args?.find()
+if (filePath) {
+    ConsumerTestUtils.runningLocally = true
+    AUXILIARY_SLEEP_MS = 1000
+
+    if (!System.getenv('CX_INFRA_IT_CF_USERNAME') || !System.getenv('CX_INFRA_IT_CF_PASSWORD')) {
+        exitPrematurely(1, 'Environment variables CX_INFRA_IT_CF_USERNAME and CX_INFRA_IT_CF_PASSWORD need to be set.')
+    }
+
+    testCaseThreads = [new TestRunnerThread(filePath)]
+
+} else {
+    ConsumerTestUtils.runningLocally = false
+
+    if (changeDoesNotNeedConsumerTesting()) {
+        notifyGithub("success", "No consumer tests necessary.")
+        exitPrematurely(0, 'No consumer tests necessary.')
+    }
+
+    /*
+    In case the build is performed for a pull request TRAVIS_COMMIT is a merge
+    commit between the base branch and the PR branch HEAD. That commit is actually built.
+    But for notifying about a build status we need the commit which is currently
+    the HEAD of the PR branch.
+
+    In case the build is performed for a simple branch (not associated with a PR)
+    In this case there is no merge commit between any base branch and HEAD of a PR branch.
+    The commit which we need for notifying about a build status is in this case simply
+    TRAVIS_COMMIT itself.
+    */
+    ConsumerTestUtils.commitHash = System.getenv('TRAVIS_PULL_REQUEST_SHA') ?: System.getenv('TRAVIS_COMMIT')
+
+    notifyGithub("pending", "Consumer tests are in progress.")
+
+    testCaseThreads = listTestCaseThreads()
+}
+
+testCaseThreads.each { it ->
+    it.start()
+}
+
+//This method will print to console while the test cases are running
+//Otherwise the job will be canceled after 10 minutes without output.
+waitForTestCases(testCaseThreads)
+
+notifyGithub("success", "All consumer tests succeeded.")
+
+
+def listTestCaseThreads() {
+    //Each dir that includes a yml file is a test case
+    def testCases = listYamlInDirRecursive(TEST_CASES_DIR)
+    def threads = []
+    testCases.each { file ->
+        threads << new TestRunnerThread(file.toString())
+    }
+    return threads
+}
+
+def waitForTestCases(threadList) {
+    threadList.metaClass.anyThreadStillAlive = {
+        for (thread in delegate) {
+            if (thread.isAlive()) {
+                return true
+            }
+        }
+        return false
+    }
+
+    def auxiliaryThread = Thread.start {
+        while (threadList.anyThreadStillAlive()) {
+            printOutputOfThreadsIfOneFailed(threadList)
+
+            sleep(AUXILIARY_SLEEP_MS)
+            if (PRINT_LOGS_AFTER_45_MINUTES_COUNTDOWN-- == 0) {
```

Comment:
We decrement each time by one. This does not mean that the logs are printed after 45 min. Convince me from the contrary. This would mean we have to decrement every millisecond. But above we sleep 10 seconds.